### ISSUE TITLE
Missing onFocus callback when the editor is focused

### ISSUE THREAD

**Issue by steven-pribilinskiy** [angular-ui/ui-ace#170]
There's a documented and available `onBlur` callback that is properly invoked when the editor is blurred.
Unfortunately there's no `onFocus` yet and there's a need for it

**Comment by sjeatdsi** [comment]
I would agree. I have multiple editors on a single page. Would be nice to know when one of them gets the focus so that I can visually indicate which editor pane is currently active by highlighting my custom header element.

**Comment by ajthinking** [comment]
Need this
